### DISCUSSION TITLE
Cleaning ~/.prefect/pg_data/

### DISCUSSION THREAD

**Discussion by Chrisprollsite** [PrefectHQ/prefect#5080] [General]
Hi,

The size of pg_data gets rapidely out of hands (~20Gb) and I was wondering if there was a way to reduce the amount of data stored to pg_data when running an agent or if there was a way to automatically clean the directory.

Thanks in advance for your help,

best,

Christian

**Comment by taljaards** [comment]
Hi @Chrisprollsite
You can refer to this topic for some tips https://discourse.prefect.io/t/how-can-i-free-up-postgres-database-space/290

**Reply by histrio** [reply]
Do you have other link with this, since it redirects me to github issues main page?

**Reply by s3v** [reply]
Hi @histrio, @Chrisprollsite 
I had to solve it myself, came up with a procedure and pg_cron job.
There might be better ways to do this (and break less of the referential integrity) but it solves my issue with disk space.
https://gist.github.com/s3v/a2cc7fa15c43a1d24b3a992659fcf698

**Reply by zzstoatzz** [reply]
thank you for sharing @s3v !!
